### PR TITLE
Read EnableWriteXorExecute from runtimeConfig

### PR DESCRIPTION
**Pull request by Quickern**
There's OS with very strict environment that forbids env variables change in some cases. And W^X can be disabled.
Added option to runtime config.

I have some concerns about configuration option name. "System.Runtime.EnableWriteXorExecute" looks very similar to other config options.

### CODE REVIEW

File: src/coreclr/utilcode/executableallocator.cpp [hunk 1]
```diff
@@ -260,7 +261,7 @@ HRESULT ExecutableAllocator::StaticInitialize(FatalErrorHandler fatalErrorHandle
 #endif
 
     g_fatalErrorHandler = fatalErrorHandler;
-    g_isWXorXEnabled = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableWriteXorExecute) != 0;
+    g_isWXorXEnabled = Configuration::GetKnobBooleanValue(W("System.Runtime.EnableWriteXorExecute"), CLRConfig::EXTERNAL_EnableWriteXorExecute);
```
Comment (mangod9) [inline src/coreclr/utilcode/executableallocator.cpp:264]: you still need to check for `!=0` here else it will default to disabled.

**Comment by mangod9** [comment]
yeah guess we will need to follow up.

File: src/coreclr/utilcode/executableallocator.cpp (hunk 1 above)
Comment (Quickern) [reply src/coreclr/utilcode/executableallocator.cpp:264]: The third argument seems like a better option, doesn't it?

    g_isWXorXEnabled = Configuration::GetKnobBooleanValue(W("System.Runtime.EnableWriteXorExecute"), CLRConfig::EXTERNAL_EnableWriteXorExecute, TRUE);

File: src/coreclr/utilcode/executableallocator.cpp [hunk 2]
```diff
@@ -260,7 +261,7 @@ HRESULT ExecutableAllocator::StaticInitialize(FatalErrorHandler fatalErrorHandle
 #endif
 
     g_fatalErrorHandler = fatalErrorHandler;
-    g_isWXorXEnabled = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableWriteXorExecute) != 0;
+    g_isWXorXEnabled = Configuration::GetKnobBooleanValue(W("System.Runtime.EnableWriteXorExecute"), CLRConfig::EXTERNAL_EnableWriteXorExecute, TRUE);
```
Comment (janvorli) [inline src/coreclr/utilcode/executableallocator.cpp:264]: I actually find the three argument version of the function a bit confusing. The thing is that there are two places where the default is defined - once at the CLRConfig::EXTERNAL_EnableWriteXorExecute definition site and once here. This version basically ignores the default at the CLRConfig::EXTERNAL_EnableWriteXorExecute definition site.
So, I'd prefer the way you had it before - without the last argument here. And I believe it was correct as it was, it didn't need any comparison to 0 as it returns `bool`.
The previous version was using the default defined at the CLRConfig::EXTERNAL_EnableWriteXorExecute definition site when neither the env var nor the runtimeConfig setting was present.

Comment (Quickern) [reply src/coreclr/utilcode/executableallocator.cpp:264]: Oh. That explains why both the code and my tests showed that all works as intended.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Review by mangod9 — APPROVED** [review]

**Comment by Quickern** [comment]
I am sorry about unnecessary command.
Any chances of backporting this PR to 8.0-staging?

@janvorli @mangod9

**Comment by mangod9** [comment]
Can you please provide a scenario/justification so we can determine feasibility to backport to 8. Thanks

**Pull request merged**